src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -523,28 +533,33 @@ internal void FindExtensionsInAssembly(ExtensionAssembly assembly)
         /// </summary>
         /// <param name="runnerAsm">The executing runner</param>
         /// <param name="extensionAsm">The extension we are attempting to load</param>
-        internal static void ValidateTargetFramework(Assembly runnerAsm, ExtensionAssembly extensionAsm)
+        internal static bool ValidateTargetFramework(Assembly runnerAsm, ExtensionAssembly extensionAsm)
         {
             if (runnerAsm == null)
-                return;
+                return true;
 
             var extHelper = new TargetFrameworkHelper(extensionAsm.FilePath);
             var runnerHelper = new TargetFrameworkHelper(runnerAsm.Location);
             if (runnerHelper.FrameworkName?.StartsWith(".NETStandard") == true)
             {
-                throw new NUnitEngineException("Test runners must target .NET Core or .NET Framework, not .NET Standard");
+                log.Warning($"{runnerAsm.FullName} test runner must target .NET Core or .NET Framework, not .NET Standard");
+                return false;
             }
             else if (runnerHelper.FrameworkName?.StartsWith(".NETCoreApp") == true)
             {
                 if (extHelper.FrameworkName?.StartsWith(".NETStandard") != true && extHelper.FrameworkName?.StartsWith(".NETCoreApp") != true)
                 {
-                    throw new NUnitEngineException(".NET Core runners require .NET Core or .NET Standard extensions");
+                    log.Warning($".NET Core runners require .NET Core or .NET Standard extension for {extensionAsm.FilePath}");
```

Comment:
Suggestion: should this and the log below be `log.Info` instead of `Warning`? I think these log lines will be hit quite regularly, and don't indicate that anything is 'wrong' - as I would expect with `log.Warning`.